src/plugins/gui/RewardsCardPlugin.tsx

```diff
@@ -56,52 +116,114 @@ export const makeRewardsCardPlugin: FiatPluginFactory = async params => {
+    const fiatCurrencyCode = wallet.fiatCurrencyCode
     const displayFiatCurrencyCode = fiatCurrencyCode.replace('iso:', '')
     showUi.enterAmount({
       headerTitle: lstrings.rewards_card_add_new_input_amount_title,
       label1: sprintf(lstrings.fiat_plugin_amount_currencycode, displayFiatCurrencyCode),
       label2: sprintf(lstrings.fiat_plugin_amount_currencycode, currencyCode),
       async onChangeText() {},
-      async onFieldChange() {
-        return undefined
+      async onFieldChange(event) {
+        const myCounter = ++counter
+
+        const { stateManager } = event
+
+        const otherFieldKey = event.value.sourceFieldNum === 1 ? 'value2' : 'value1'
+        const spinnerKey = event.value.sourceFieldNum === 1 ? 'spinner2' : 'spinner1'
+
+        stateManager.update({ [spinnerKey]: true })
+
+        if (eq(event.value.value, '0')) {
+          stateManager.update({ [otherFieldKey]: '', [spinnerKey]: false })
+          return
+        }
+        let quoteParams: FiatProviderGetQuoteParams
+
+        if (event.value.sourceFieldNum === 1) {
+          // User entered a fiat value. Convert to crypto
+          quoteParams = {
+            wallet,
+            pluginId: wallet.currencyInfo.pluginId,
+            displayCurrencyCode: currencyCode,
+            exchangeAmount: event.value.value,
+            fiatCurrencyCode,
+            amountType: 'fiat',
+            ...redundantQuoteParams
+          }
+        } else {
+          // User entered a crypto value. Convert to fiat
+          quoteParams = {
+            wallet,
+            pluginId: wallet.currencyInfo.pluginId,
+            displayCurrencyCode: currencyCode,
+            exchangeAmount: event.value.value,
+            fiatCurrencyCode,
+            amountType: 'crypto',
+            ...redundantQuoteParams
+          }
+        }
+
+        const bestQuote = await ioniaProvider.getQuote(quoteParams).catch(error => {
+          stateManager.update({ statusText: { content: String(error), textType: 'error' }, [spinnerKey]: false })
+          throw error
+        })
+
+        // Abort to avoid race conditions
+        if (myCounter !== counter) return
+
+        const exchangeRateText = getRateFromQuote(bestQuote, displayFiatCurrencyCode)
+        stateManager.update({
+          statusText: { content: exchangeRateText },
+          // poweredBy: { poweredByText: bestQuote.pluginDisplayName, poweredByIcon: bestQuote.partnerIcon },
+          [otherFieldKey]: event.value.sourceFieldNum === 1 ? toFixed(bestQuote.cryptoAmount, 0, 6) : toFixed(bestQuote.fiatAmount, 0, 2),
+          [spinnerKey]: false
+        })
       },
       async onPoweredByClick() {},
       async onSubmit(event) {
-        const spendInfo: EdgeSpendInfo = {
-          currencyCode,
-          spendTargets: [
-            {
-              nativeAmount: event.value.response.value2
-            }
-          ]
+        const fiatAmount = parseFloat(event.value.response.value1)
```

Comment:
All of this logic should be in the provider. The plugin shouldn't have ANY knowledge of how the payment is made. Instead just call `quote.approveQuote`